nucypher/blockchain/eth/signers.py

```diff
@@ -191,3 +216,93 @@ def unlock_account(self, account: str, password: str, duration: int = None) -> b
     @validate_checksum_address
     def lock_account(self, account: str):
         return True
+
+
+class KeyStoreSigner(Signer):
+
+    IS_SIGNER_LOCAL = True
+
+    def __init__(self, uri: str):
+        super().__init__()
+
+        decoded_uri = urlparse(uri)
+        if decoded_uri.scheme not in ('key', 'keystore') or decoded_uri.netloc:
+            raise cls.InvalidSignerURI(uri)
+
+        with open(decoded_uri.path) as file_obj:
+            key = json.load(file_obj)
+
+        address = key['address']
+        if not Web3.isAddress(address):
+            raise ValueError(f"'{decoded_uri.path}' does not contain a valid address")
+
+        self.key = key
+        self._account = Web3.toChecksumAddress(address)
+        self._signer = None
+
+    def _is_my_account(self, account) -> bool:
+        return Web3.isAddress(account) and \
+            self.account == Web3.toChecksumAddress(account)
+
+    @classmethod
+    def from_signer_uri(cls, uri: str) -> 'Signer':
+        return cls(uri)
+
+    @property
+    def account(self) -> str:
+        return self._account
+
+    @validate_checksum_address
+    def is_device(self, account: str) -> bool:
+        return False
+
+    def accounts(self) -> List[str]:
+        return [self.account]
```

Comment:
Interesting moment here - To live up to it;s name What do you think about the `KeystoreSigner` being able to manage the the keystore *directory*, reading the private key into memory just-in-time?  This will be a bit more consistent with other signing providers.

Alternately - IMO it's okay to disable all other account management if using a `KeystoreSigner`.

Reply:
Actually supporting only a single file and requiring its name kind of sucks, especially as the keystore file names that geth uses are incredibly long and vary depending on address.

Multiple account support it is then (#1855 was half way there too).

Reply:
Does supporting a keystore dir even make sense if we only allow this for Usula?

Reply:
Ok, thanks for the comments and issues raised - the current commit on this PR branch should hopefully address all of them adequately and be suitable as usable single-key-store-file implementation for the worker.

Adding multi-file support is not trivial but not a big stretch and could be made backward compatible - perhaps this is best left to another PR.